neuralmonkey/decoding_function.py

```diff
@@ -24,15 +25,17 @@ def __init__(self, attention_states, scope, input_weights=None,
             scope: The name of the variable scope in the graph used by this
                    attention object.
             input_weights: (Optional) The padding weights on the input.
-            max_fertility: (Optional) For the Coverage attention compatibilty,
-                           maximum fertility of one word.
+            attention_fertility: (Optional) For the Coverage attention
+                compatibilty, maximum fertility of one word.
+            runtime_mode: (Optional) Indicates whether the object will be used
+                          for runtime decoding.
         """
         self.scope = scope
         self.attentions_in_time = []
         self.attention_states = attention_states
         self.input_weights = input_weights
 
-        with tf.variable_scope(scope):
+        with tf.variable_scope(scope, reuse=runtime_mode):
```

Comment:
A je tam vůbec nutný tohelto reuse? Když se řekne reuse o úroveň výš, tak se to přenáší i na ty vnitřní scopy, ne?
Jinak mi přijde, že by to nikdy nemělo vyrobit nový proměnný, ale vždycky spadnout.
Jak by měly vypadat ty asserty, nic rozumného mě nenapadá.